### ISSUE TITLE
[BUG] Snapshot wires out of order

### ISSUE THREAD

**Issue by jsmallz333** [PennyLaneAI/pennylane#6427]
### Expected behavior

The order of the wires returned by a `qml.Snapshot()` should match those of the full execution

### Actual behavior

The wires are out of order, based on the order in which they had operations applied to them.

### Additional information

In the example I provide, I include a commented line that may be uncommented for an easy workaround, s.t. the snapshot's order matches that of the 'execution_result'

I found that a 2-wire example did not duplicate the issue, but returned the correct wire order in the snapshots.

I have not debugged these, but it seems that this may arise from a difference between lines 194 and 203 of `pennylane/debugging/snapshot.py` in the implementation of the measurements with a `wire_order`/`wire_map`

### Source code

```shell
@qml.qnode(qml.device("default.qubit", wires=3))
def circuit():
    #[qml.Identity(wires=x) for x in range(3)]
    qml.Hadamard(wires=2)
    qml.Snapshot("hadamard on 2")
    qml.Hadamard(wires=0)
    qml.Hadamard(wires=0)
    qml.Snapshot()
    return qml.state()
qml.snapshots(circuit)()
```

### Tracebacks

```shell
{'hadamard on 2': tensor([0.70710678+0.j, 0.        +0.j, 0.70710678+0.j, 0.        +0.j], requires_grad=True),
 1: tensor([0.70710678+0.j, 0.        +0.j, 0.70710678+0.j, 0.        +0.j], requires_grad=True),
 'execution_results': tensor([0.70710678+0.j, 0.70710678+0.j, 0.        +0.j, 0.        +0.j,
         0.        +0.j, 0.        +0.j, 0.        +0.j, 0.        +0.j], requires_grad=True)}
```

### System information

```shell
Name: PennyLane
Version: 0.38.0
Summary: PennyLane is a cross-platform Python library for quantum computing, quantum machine learning, and quantum chemistry. Train a quantum computer the same way as a neural network.
Home-page: https://github.com/PennyLaneAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /home/jsmalley/qdrive/anaconda3/envs/penny2/lib/python3.9/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, packaging, pennylane-lightning, requests, rustworkx, scipy, toml, typing-extensions
Required-by: PennyLane-Catalyst, PennyLane-qiskit, PennyLane_Lightning

Platform info:           Linux-6.5.0-1025-azure-x86_64-with-glibc2.35
Python version:          3.9.18
Numpy version:           1.26.3
Scipy version:           1.11.4
Installed devices:
- default.clifford (PennyLane-0.38.0)
- default.gaussian (PennyLane-0.38.0)
- default.mixed (PennyLane-0.38.0)
- default.qubit (PennyLane-0.38.0)
- default.qubit.autograd (PennyLane-0.38.0)
- default.qubit.jax (PennyLane-0.38.0)
- default.qubit.legacy (PennyLane-0.38.0)
- default.qubit.tf (PennyLane-0.38.0)
- default.qubit.torch (PennyLane-0.38.0)
- default.qutrit (PennyLane-0.38.0)
- default.qutrit.mixed (PennyLane-0.38.0)
- default.tensor (PennyLane-0.38.0)
- null.qubit (PennyLane-0.38.0)
- lightning.qubit (PennyLane-Lightning-0.38.0)
- qiskit.aer (PennyLane-qiskit-0.36.0)
- qiskit.basicaer (PennyLane-qiskit-0.36.0)
- qiskit.basicsim (PennyLane-qiskit-0.36.0)
- qiskit.ibmq (PennyLane-qiskit-0.36.0)
- qiskit.ibmq.circuit_runner (PennyLane-qiskit-0.36.0)
- qiskit.ibmq.sampler (PennyLane-qiskit-0.36.0)
- qiskit.remote (PennyLane-qiskit-0.36.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by CatalinaAlbornoz** [comment]
Thanks for reporting this @jsmallz333 !

**Comment by albi3ro** [comment]
Marking this as closed by #6461
